addons/point_of_sale/models/pos_session.py

```diff
@@ -2099,6 +2109,118 @@ def get_pos_ui_product_pricelist_item_by_product(self, product_tmpl_ids, product
             ('product_id', 'in', product_ids)]
         return self.env['product.pricelist.item'].search_read(pricelist_item_domain, self._product_pricelist_item_fields())
 
+    def _is_capture_system_activated(self):
+        # Enabled by default, but can be disabled manually if needed (error, crashs, storage issue due to number of attachments)
+        return self.env['ir.config_parameter'].sudo().get_param('point_of_sale.capture_unprocessed_order',
+                                                                'true').lower() == 'true'
+
+    def _handle_order_process_fail(self, order: dict, exception: Exception, draft: bool):
+        if not self._is_capture_system_activated():
+            return
+
+        if draft:
+            # draft will be set when we receive a restaurant order that was not validated yet
+            # if we capture it, it will create tons of (a priori) irrelevant attachments
+            # So we capture only restaurant order that are validated
+            _logger.info("order '%s' was not captured as it is draft", order['data']['name'])
+            return
+
+        # Create records into a dedicated cursor (as the current one will roll back when re-raising the exception)
+        with registry().cursor() as cr:
+            env = api.Environment(
+                cr,
+                SUPERUSER_ID,  # Impersonate odoobot on purpose to bypass permissions on pos.session logging
+                {}
+            )
+            self.with_env(env)._process_order_process_fail(order, exception, self.env.user.id) # purposefully use the original user (and not odoobot)
+
+    def _process_order_process_fail(self, order: dict, exception: Exception, uid: int = False):
+        current_pos_order_data_hash = hash(PoSOrderData.from_data(order['data']))
+        self._get_unprocessed_pos_order_scheduled_activity(
+            order['data']['name'],
+            current_pos_order_data_hash,
+            uid or self.env.user.id,
+            create=True,
+        )
+        self._capture_order_data(order, exception, current_pos_order_data_hash)
+
+    def _shorten_pos_order_data_hash(self, pos_order_data_hash: int):
+        # The bigger the value, the less likely a hash collision will occur
+        # but a bigger value also means a longer name in the attachment
+        return str(pos_order_data_hash)[:6]
```

Comment:
is 6 characters safe for collisions? than means only one million possibilites

Reply:
the case should be pretty rare, so I think it is enough